### PR TITLE
[CI] Upgrade ci tools

### PR DESCRIPTION
**Pull request by AzHicham**
Upgraded tools: 
pre-commit: 1.18.3 -> 2.9.2
black: 18.9b0 -> 22.3.0
clang-format: 6.0 -> 10.0 (@hove-io/coreteam please run apt install clang-format to install clang-format-10 on ubuntu-20.04)
sonar-cli: 4.6.0.2311 -> 4.7.0.2747 
mypy: v0.670 -> v0.800

black, mypy, have been deleted from requierement-pre-commit.txt because they must be installed with the following command: pre-commit install --install-hooks

### CODE REVIEW

File: source/jormungandr/tests/bragi_autocomplete_tests.py [hunk 1]
```diff
@@ -786,7 +785,7 @@ def test_autocomplete_call_with_param_places_proximity_radius(self):
             assert params.get('proximity_decay') == ['0.4']
 
     def test_autocomplete_call_override(self):
-        """"
+        """ "
```
Comment (woshilapin) [inline source/jormungandr/tests/bragi_autocomplete_tests.py]: Probably a useless extra double-quotes here.

File: source/jormungandr/tests/departure_board_tests.py [hunk 2]
```diff
@@ -1505,7 +1505,7 @@ class TestFirstLastDatetimeWithNegativeTimezone(AbstractTestFixture):
     """
 
     def test_first_last_datetime_same_day(self):
-        """              20170102
+        """20170102
```
Comment (woshilapin) [inline source/jormungandr/tests/departure_board_tests.py]: I'd say the extra spaces here are needed to align the text with these ASCII arts (see the entire file for similar correction).

**Review by woshilapin — APPROVED** [review]

**Review by pbench — APPROVED** [review]

File: source/jormungandr/tests/bragi_autocomplete_tests.py [hunk 3]
```diff
@@ -706,8 +706,7 @@ def mock_bragi_autocomplete_call(bragi_response, limite=10, http_response_code=2
 @dataset({'main_routing_test': MOCKED_INSTANCE_CONF}, global_config={'activate_bragi': True})
 class TestBragiAutocomplete(AbstractTestFixture):
     def test_autocomplete_no_json(self):
-        """
-        """
+        """ """
```
Comment (woshilapin) [inline source/jormungandr/tests/bragi_autocomplete_tests.py]: Hmm, this one look weirds too :-D

File: source/jormungandr/jormungandr/stat_manager.py [hunk 4]
```diff
@@ -296,7 +296,7 @@ def fill_error(self, stat_request, error):
             stat_error.message = error['message']
 
     def fill_journey(self, stat_journey, resp_journey):
-        """"
+        """ "
```
Comment (woshilapin) [inline source/jormungandr/jormungandr/stat_manager.py]: Here is an extra double quote too.

File: source/jormungandr/tests/bragi_autocomplete_tests.py (hunk 3 above)
Comment (AzHicham) [reply source/jormungandr/tests/bragi_autocomplete_tests.py]: already fixed :)

File: source/jormungandr/jormungandr/stat_manager.py (hunk 4 above)
Comment (AzHicham) [reply source/jormungandr/jormungandr/stat_manager.py]: already fixed :)

**Pull request merged**